src/pool/srv_target.c

```diff
@@ -1263,9 +1261,17 @@ ds_pool_stop(uuid_t uuid)
 
 	ds_rebuild_abort(pool->sp_uuid, -1, -1, -1);
 	ds_migrate_stop(pool, -1, -1);
+
 	ds_pool_put(pool); /* held by ds_pool_start */
-	pool_put_sync(pool);
-	D_INFO(DF_UUID": pool stopped\n", DP_UUID(uuid));
+
+	while (!daos_lru_is_last_user(&pool->sp_entry))
+		dss_sleep(1000 /* ms */);
```

Comment:
I don't see why we have to wait for the ds_pool reference dropped here, it's better to print some warning messages if it hangs here due to potential defects.

Reply:
I see. I don't think currently any user expect that, the ds_pool_child could be stopped due to faulty target at any time. Anyway, let's see if any potential ds_pool reference issue could be discovered by this new change.

Reply:
I have indeed told Fan Yong that nowadays `ds_pool_child` may be stopped individually.

Reply:
I'm reluctant to log repeated `D_INFO` messages here, so only added a completion message so that we know if we are blocked in the reference wait or `pool_child_delete_all`.

Reply:
@Nasf-Fan, active cancellation seems to be a large project. It is out of the scope of this PR, but has been in my plan for a long time.

"That" is not enough. Without this PR, when we wait for `ds_pool_child` references, we are in `pool_free_ref`, which is called after the `ds_pool` object has been deleted from the LRU cache. Hence a retried pool destroy operation will not find the `ds_pool` object. Please see the PR description. (I began this PR with only such a change, but testing proved it to be insufficient.) On the other hand, stopping activities when freeing the last reference is not a good model, so I plan to move that out anyway.

Reply:
I have no better idea. Then let's resolve the potential issues when really hit in the future.